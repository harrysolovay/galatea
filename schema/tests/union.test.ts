import { assertSnapshot } from "@std/testing/snapshot"
import * as T from "../mod.ts"

Deno.test("No Variant Values", async (t) => {
  const Inner = T.union({ A: T.none, B: T.none })
  const Root = T.object({ inner: Inner })
  await assertSnapshot(t, T.schema({ Root, Inner }, "Root"))
})

Deno.test("Variant Values", async (t) => {
  const Inner = T.union({ A: T.string, B: T.number, C: T.none })
  const Root = T.object({ inner: Inner })
  await assertSnapshot(t, T.schema({ Root, Inner }, "Root"))
})
