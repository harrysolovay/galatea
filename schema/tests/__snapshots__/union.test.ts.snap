export const snapshot = {};

snapshot[`No Variant Values 1`] = `
{
  "\$defs": {
    Inner: {
      oneOf: [
        {
          additionalProperties: false,
          properties: {
            type: {
              const: "A",
            },
          },
          required: [
            "type",
          ],
        },
        {
          additionalProperties: false,
          properties: {
            type: {
              const: "B",
            },
          },
          required: [
            "type",
          ],
        },
      ],
    },
    Root: {
      additionalProperties: false,
      properties: {
        inner: {
          "\$ref": "#/\$defs/Inner",
        },
      },
      required: [
        "inner",
      ],
      type: "object",
    },
  },
  additionalProperties: false,
  properties: {
    inner: {
      "\$ref": "#/\$defs/Inner",
    },
  },
  required: [
    "inner",
  ],
  type: "object",
}
`;

snapshot[`Variant Values 1`] = `
{
  "\$defs": {
    Inner: {
      oneOf: [
        {
          additionalProperties: false,
          properties: {
            type: {
              const: "A",
            },
            value: {
              type: "string",
            },
          },
          required: [
            "type",
            "value",
          ],
        },
        {
          additionalProperties: false,
          properties: {
            type: {
              const: "B",
            },
            value: {
              type: "number",
            },
          },
          required: [
            "type",
            "value",
          ],
        },
        {
          additionalProperties: false,
          properties: {
            type: {
              const: "C",
            },
          },
          required: [
            "type",
          ],
        },
      ],
    },
    Root: {
      additionalProperties: false,
      properties: {
        inner: {
          "\$ref": "#/\$defs/Inner",
        },
      },
      required: [
        "inner",
      ],
      type: "object",
    },
  },
  additionalProperties: false,
  properties: {
    inner: {
      "\$ref": "#/\$defs/Inner",
    },
  },
  required: [
    "inner",
  ],
  type: "object",
}
`;
